cli/internal/telemetry/telemetry.go

```diff
@@ -0,0 +1,102 @@
+package telemetry
+
+import (
+	"fmt"
+	"os"
+	"sync"
+
+	"github.com/google/uuid"
+	"github.com/rudderlabs/analytics-go/v4"
+	"github.com/rudderlabs/rudder-iac/cli/internal/config"
+)
+
+var (
+	once              sync.Once
+	v                 string
+	telemetryDisabled bool
+)
+
+func getEnvWithFallback(key, fallback string) string {
+	if value := os.Getenv(key); value != "" {
+		return value
+	}
+	return fallback
+}
+
+func Initialise(version string) {
+	once.Do(func() {
+		conf := config.GetConfig()
+
+		v = version
+		if conf.Telemetry.Disabled {
+			telemetryDisabled = true
+			return
+		}
+
+		disabled := getEnvWithFallback("RUDDER_CLI_TELEMETRY_DISABLED", "")
+		if disabled != "" && disabled == "1" {
+			telemetryDisabled = true
+			return
+		}
+
+		if conf.Telemetry.UserID == "" {
+			userID := uuid.New().String()
+			config.SetTelemetryUserID(userID)
+		}
+	})
+}
+
+func DisableTelemetry() {
+	telemetryDisabled := true
+	config.SetTelemetryDisabled(telemetryDisabled)
+}
+
+func EnableTelemetry() {
+	telemetryDisabled := false
+	config.SetTelemetryDisabled(telemetryDisabled)
+}
+
+func track(event string, properties analytics.Properties) error {
+	if telemetryDisabled {
+		return nil
+	}
+
+	var (
+		writeKey     = config.GetTelemetryWriteKey()
+		dataplaneURL = config.GetTelemetryDataplaneURL()
+	)
+
+	client, err := analytics.NewWithConfig(writeKey, analytics.Config{
+		DataPlaneUrl: dataplaneURL,
+		Logger:       NewTelemetryLogger(),
+	})
+
+	if err != nil {
+		return fmt.Errorf("failed to create analytics client: %w", err)
+	}
+
+	defer client.Close()
```

Comment:
Shouldn't we be having a singleton client and use that instead of starting a new client on every track call?

Reply:
I did think about this but in the end for now we are tracking a single command's result, so atmost we will be firing a single track call per execution. That was the understanding and also to keep things simple, went with a client created per call.

Reply:
Cool, we can revisit if we need to introduce multiple events per command at some point.